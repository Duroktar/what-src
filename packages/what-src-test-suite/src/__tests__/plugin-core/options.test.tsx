import * as T from '@what-src/plugin-core'

describe('what-src configuration', () => {
  const service = T.getService()
  const options = service.options

  it('has stable output', () => {
    expect<T.WhatSrcConfiguration>(options).toMatchSnapshot()
  })

  it('has default options', () => {
    expect(options).toEqual(T.defaultOptions)
  })

  it('doesn\'t run in production mode by default', () => {
    expect(options.productionMode).toBeFalsy()
  })

  it('is configurable', () => {
    const configurableService = T.getService({
      options: {
        productionMode: true,
      },
    })

    const customOptions = configurableService.options
    customOptions.serverUrl = 'new_server_url'
    expect(customOptions.serverUrl).toEqual('new_server_url')
    expect(customOptions.productionMode).toEqual(true)

    expect<T.WhatSrcConfiguration>(customOptions).toMatchSnapshot()
  })
})
