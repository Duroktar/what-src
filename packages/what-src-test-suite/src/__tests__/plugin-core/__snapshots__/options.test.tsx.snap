// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`what-src configuration has stable output 1`] = `
Object {
  "baseDirOverride": null,
  "blacklistedTags": Array [
    "Helmet",
    "head",
    "meta",
    "script",
  ],
  "cacheFileName": "whatSrcRuntime.js",
  "cacheLocOverride": null,
  "createCacheDir": true,
  "createCacheFile": true,
  "dataTag": "data-what-src",
  "enableXkcdMode": false,
  "globalCacheKey": "__what-src-global-callback-key",
  "preventDefault": true,
  "productionMode": false,
  "serverUrl": "http://localhost:8018/__what_src",
  "stopPropagation": true,
  "useRemote": false,
}
`;

exports[`what-src configuration is configurable 1`] = `
Object {
  "baseDirOverride": null,
  "blacklistedTags": Array [
    "Helmet",
    "head",
    "meta",
    "script",
  ],
  "cacheFileName": "whatSrcRuntime.js",
  "cacheLocOverride": null,
  "createCacheDir": true,
  "createCacheFile": true,
  "dataTag": "data-what-src",
  "enableXkcdMode": false,
  "globalCacheKey": "__what-src-global-callback-key",
  "preventDefault": true,
  "productionMode": true,
  "serverUrl": "new_server_url",
  "stopPropagation": true,
  "useRemote": false,
}
`;
